src/pages/index.js

```diff
@@ -53,67 +54,206 @@ newCardPopup.setEventListeners();
 const popupWithImage = new PopupWithImage("#image-modal");
 popupWithImage.setEventListeners();
 
+// new popup for deleting a card
+const deleteCardPopup = new PopupWithConfirmation(
+  "#delete-card-modal",
+  handleDeleteCard
+);
+// set event listeners for delete card popup
+deleteCardPopup.setEventListeners();
+
+// new popup for changing avatar
+const changeAvatarPopup = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarSubmit
+);
+changeAvatarPopup.setEventListeners();
+
+// =========  USER INFO  ========= //
+
 // Instance of User Info
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-// Instance of Section
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
+// =========  APIs  ========= //
+
+// Instance of Api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d918969b-47fa-4f72-8138-61be6ab55840",
+    "Content-Type": "application/json",
   },
-  "#cards__list"
-);
+});
 
-section.renderItems();
-/////////////////////////////////////////////////////////////////////
+// fetch and render initial cards
+api
+  .getInitialCards()
+  .then((result) => {
+    // Instance of Section
+    let section = new Section(
+      {
+        items: result.reverse(),
+        renderer: createCard,
+      },
+      "#cards__list"
+    );
+    section.renderItems(result);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
 
-// FUNCTIONS
+// fetch and render user info
+api
+  .getUserInfo()
+  .then((result) => {
+    // update UI to reflect the user info from the db
+    // console.log(result);
+    userInfo.setUserInfo(result.name, result.about);
+    userInfo.setAvatar(result.avatar);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+//////////////////////////////////////////////////////////
+
+// =========  FUNCTIONS  ========= //
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick,
+    handleUnlikeClick
+  );
   return card.getView();
 }
 
-//render card
 function renderCard(cardData) {
   const cardView = createCard(cardData);
+  const section = new Section({}, "#cards__list");
   section.addItem(cardView);
 }
 
-// declare a function to handle image click
 function handleImageClick({ name, link }) {
+  // open the popup with the image
   popupWithImage.open({ link, name });
 }
 
-// EVENT HANDLERS
+// =========  EVENT HANDLERS  ========= //
 
 // Handles edit form submit
-function handleProfileEditSubmit() {
-  editFormValidator.disableSubmitButton();
-  editProfilePopup.close();
+function handleProfileEditSubmit({ title, description }) {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
Thank you for the suggestion, I'll work on this implementation at a later time.  I need to move on for now :)